File: 3033.modify-the-matrix.cs
/*
 * @lc app=leetcode id=3033 lang=csharp
 *
 * [3033] Modify the Matrix
 */

// @lc code=start
using System;
using System.Collections.Generic;

public class Solution {
    public int[][] ModifiedMatrix(int[][] matrix) {
        int m = matrix.Length;
        int n = matrix[0].Length;
        for (int j = 0; j < n; j++) {
            int maxEle = -1;
            List<int> negOnes = new List<int>();
            for (int i = 0; i < m; i++) {
                maxEle = Math.Max(maxEle, matrix[i][j]);
                if (matrix[i][j] == -1) negOnes.Add(i);
            }
            foreach (int i in negOnes) matrix[i][j] = maxEle;
        }
        return matrix;
    }
}

// @lc code=end

File: 3033.modify-the-matrix.py
#
# @lc app=leetcode id=3033 lang=python3
#
# [3033] Modify the Matrix
#

# @lc code=start
from typing import List


class Solution:
    def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:
        m, n = len(matrix), len(matrix[0])
        for j in range(n):
            maxEle = -1
            negOnes = []
            for i in range(m):
                maxEle = max(maxEle, matrix[i][j])
                if matrix[i][j] == -1: negOnes.append(i)
            for i in negOnes: matrix[i][j] = maxEle
        return matrix
# @lc code=end

